ast/src/expression/binary.rs

```diff
@@ -16,33 +16,59 @@
 
 use super::*;
 
+/// A binary operator.
+///
+/// Precedence is defined in the parser.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum BinaryOperation {
+    /// Addition, i.e. `+`.
     Add,
+    /// Subtraction, i.e. `-`.
     Sub,
+    /// Multiplication, i.e. `*`.
     Mul,
+    /// Division, i.e. `/`.
     Div,
+    /// Exponentiation, i.e. `**` in `a ** b`.
     Pow,
+    /// Logical-or, i.e., `||`.
     Or,
+    /// Logical-and, i.e., `&&`.
     And,
+    /// Equality relation, i.e., `==`.
     Eq,
+    /// In-equality relation, i.e. `!=`.
     Ne,
+    /// Greater-or-equal relation, i.e. `>=`.
     Ge,
+    /// Greater-than relation, i.e. `>=`.
     Gt,
+    /// Lesser-or-equal relation, i.e. `<=`.
     Le,
+    /// Lesser-than relation, i.e. `<`.
     Lt,
+    /// Bitwise-or, i.e., `|`.
```

Comment:
Should say inclusive. If taking suggestion above, this could be 'bitwise inclusive disjunction'.